modules/internal/DefaultSparse.chpl

```diff
@@ -406,6 +513,21 @@ module DefaultSparse {
       return irv;
     }
 
+    proc sparseBulkShiftArray(shiftMap, oldnnz){
```

Comment:
It's probably obvious to you what sparseBulkShiftArray is doing and why, but I don't understand immediately. Could you add a comment to it describing what it does here? I think it would be also nice to have a comment describing sparseShiftArray. I don't already know what either of these do...